From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 15 Mar 2018 14:12:54 -0400
Subject: [PATCH] Fix the efidir in grub-setpassword

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 util/grub-setpassword.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util/grub-setpassword.in b/util/grub-setpassword.in
index d7924af5192..cf70257eed6 100644
--- a/util/grub-setpassword.in
+++ b/util/grub-setpassword.in
@@ -1,7 +1,8 @@
 #!/bin/sh -e
 
+EFIDIR=$(grep ^ID= /etc/os-release | sed -e 's/^ID=//' -e 's/rhel/redhat/')
 if [ -d /sys/firmware/efi/efivars/ ]; then
-    grubdir=`echo "/@bootdirname@/efi/EFI/redhat/" | sed 's,//*,/,g'`
+    grubdir=`echo "/@bootdirname@/efi/EFI/${EFIDIR}/" | sed 's,//*,/,g'`
 else
     grubdir=`echo "/@bootdirname@/@grubdirname@" | sed 's,//*,/,g'`
 fi
