From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 27 Aug 2018 13:10:08 -0400
Subject: [PATCH] Mark some unused stuff unused

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/commands/blscfg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grub-core/commands/blscfg.c b/grub-core/commands/blscfg.c
index bdb1c5a95aa..abd6f00d0de 100644
--- a/grub-core/commands/blscfg.c
+++ b/grub-core/commands/blscfg.c
@@ -434,7 +434,7 @@ finish:
 
 static grub_envblk_t saved_env = NULL;
 
-static int
+static int UNUSED
 save_var (const char *name, const char *value, void *whitelist UNUSED)
 {
   const char *val = grub_env_get (name);
@@ -446,7 +446,7 @@ save_var (const char *name, const char *value, void *whitelist UNUSED)
   return 0;
 }
 
-static int
+static int UNUSED
 unset_var (const char *name, const char *value UNUSED, void *whitelist)
 {
   grub_dprintf("blscfg", "restoring \"%s\"\n", name);
